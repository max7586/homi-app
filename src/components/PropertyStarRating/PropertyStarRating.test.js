import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {PropertyStarRating} from './PropertyStarRating';
import { BrowserRouter } from 'react-router-dom'

describe('PropertyStarRating component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <BrowserRouter>
        <PropertyStarRating />
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<PropertyStarRating  rating={5}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('renders the UI as expected with no rating', () => {
    const tree = renderer
      .create(<PropertyStarRating rating={0}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});



