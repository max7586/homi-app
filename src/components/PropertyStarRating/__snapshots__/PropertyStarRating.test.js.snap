// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyStarRating component renders the UI as expected 1`] = `
<span
  className="PropertyStarRating"
/>
`;

exports[`PropertyStarRating component renders the UI as expected with no rating 1`] = `
<span
  className="PropertyStarRating"
/>
`;
